hazelcast/src/main/java/com/hazelcast/map/impl/event/MapEventPublishingService.java

```diff
@@ -124,7 +124,7 @@ private void dispatchEntryEventData(EntryEventData entryEventData, ListenerAdapt
     private Member getMember(EventData eventData) {
         Member member = nodeEngine.getClusterService().getMember(eventData.getCaller());
         if (member == null) {
-            member = new MemberImpl(eventData.getCaller(), false);
+            member = new MemberImpl(eventData.getCaller(), nodeEngine.getNode().getVersion(), false);
```

Comment:
Shouldn't node's codebase version be available on `NodeEngine` interface?  `NodeEngine.getVersion()`? So, no need to cast impl...